OracleWebLogic/samples/12213-domain-home-in-image-ssl/Dockerfile

```diff
@@ -0,0 +1,83 @@
+#Copyright (c) 2014-2018 Oracle and/or its affiliates. All rights reserved.
+#
+#Licensed under the Universal Permissive License v 1.0 as shown at http://oss.oracle.com/licenses/upl.
+#
+# ORACLE DOCKERFILES PROJECT
+# --------------------------
+# This Dockerfile extends the Oracle WebLogic image by creating a sample domain.
+#
+# Util scripts are copied into the image enabling users to plug NodeManager
+# automatically into the AdminServer running on another container.
+#
+# HOW TO BUILD THIS IMAGE
+# -----------------------
+# Put all downloaded files in the same directory as this Dockerfile
+# Run:
+#      $ sudo docker build -t 12213-domain-home-image
+#
+# Pull base image
+# ---------------
+FROM  container-registry.oracle.com/middleware/weblogic:12.2.1.3
+
+# Maintainer
+# ----------
+MAINTAINER Monica Riccelli <monica.riccelli@oracle.com>
+
+ARG CUSTOM_DOMAIN_NAME="${CUSTOM_DOMAIN_NAME:-domain1}"
+ARG CUSTOM_ADMIN_PORT="${CUSTOM_ADMIN_PORT:-7001}"  
+ARG CUSTOM_ADMIN_SERVER_SSL_PORT="${CUSTOM_ADMIN_SERVER_SSL_PORT:-7002}"  
+ARG CUSTOM_MANAGED_SERVER_PORT="${CUSTOM_MANAGED_SERVER_PORT:-8001}"
+ARG CUSTOM_MANAGED_SERVER_SSL_PORT="${CUSTOM_MANAGED_SERVER_SSL_PORT:-8002}"
+ARG CUSTOM_DEBUG_PORT="${CUSTOM_DEBUG_PORT:-8453}"
+ARG CUSTOM_ADMIN_NAME="${CUSTOM_ADMIN_NAME:-admin-server}"
+ARG CUSTOM_ADMIN_HOST="${CUSTOM_ADMIN_HOST:-wlsadmin}"
+ARG CUSTOM_CLUSTER_NAME="${CUSTOM_CLUSTER_NAME:-DockerCluster}"
+ARG CUSTOM_SSL_ENABLED="${CUSTOM_SSL_ENABLED:-true}"
+
+# WLS Configuration
+# ---------------------------
+ENV ORACLE_HOME=/u01/oracle \
+    PROPERTIES_FILE_DIR="/u01/oracle/properties" \
+    SSL_ENABLED="${CUSTOM_SSL_ENABLED}" \
+    DOMAIN_NAME="${CUSTOM_DOMAIN_NAME}" \
+    DOMAIN_HOME="/u01/oracle/user_projects/domains/${CUSTOM_DOMAIN_NAME}" \
+    ADMIN_PORT="${CUSTOM_ADMIN_PORT}" \
+    ADMIN_SERVER_SSL_PORT="${CUSTOM_ADMIN_SERVER_SSL_PORT}" \
+    ADMIN_NAME="${CUSTOM_ADMIN_NAME}" \
+    ADMIN_HOST="${CUSTOM_ADMIN_HOST}" \
+    CLUSTER_NAME="${CUSTOM_CLUSTER_NAME}" \
+    MANAGED_SERVER_PORT="${CUSTOM_MANAGED_SERVER_PORT}" \
+    MANAGED_SERVER_SSL_PORT="${CUSTOM_MANAGED_SERVER_SSL_PORT}" \
+    MANAGED_SERV_NAME="${MANAGED_SERV_NAME}" \
+    DEBUG_PORT="8453" \
+    PATH=$PATH:/u01/oracle/oracle_common/common/bin:/u01/oracle/wlserver/common/bin:${DOMAIN_HOME}:${DOMAIN_HOME}/bin:/u01/oracle
+
+# Add files required to build this image
+COPY --chown=oracle:oracle container-scripts/* /u01/oracle/
```

Comment:
Are the scripts in this demo actually different to the upstream ones? Can't we incorporate these changes into the upstream base image so that it's just a boolean option to enable or disable HTTPS?